scripts/apps/authoring/authoring/directives/ItemAssociationDirective.js

```diff
@@ -278,15 +293,31 @@ export function ItemAssociationDirective(superdesk, renditions, config, authorin
             scope.upload = function() {
                 if (scope.editable) {
                     superdesk.intent('upload', 'media', {
-                        uniqueUpload: true,
+                        uniqueUpload: scope.maxUploads === undefined || scope.maxUploads === 1,
+                        maxUploads: scope.maxUploads,
                         allowPicture: scope.allowPicture,
                         allowVideo: scope.allowVideo,
                         allowAudio: scope.allowAudio
                     }).then((images) => {
                         // open the view to edit the PoI and the cropping areas
                         if (images) {
                             scope.$applyAsync(() => {
-                                scope.edit(images[0]);
+                                var [rootField, index] = mediaIdGenerator.getFieldParts(scope.rel);
+                                var imagesWithIds = [];
+
+                                function editNextFile() {
+                                    if (imagesWithIds.length > 0) {
+                                        var imageWithId = imagesWithIds.shift();
+
+                                        scope.edit(imageWithId.image, imageWithId.id, editNextFile);
+                                    }
+                                }
+
+                                _.forEach(images, (image) => {
```

Comment:
do we actually need user to edit all images right after uploading them?

Reply:
This was the flow when you uploaded one image

Reply:
This was how the functionality was implemented. I guess we should ask if we want to change that, but I'd do that in another PR.

Reply:
I see